h3/src/connection.rs

```diff
@@ -826,20 +828,18 @@ where
                 //# The MAX_PUSH_ID frame is always sent on the control stream.  Receipt
                 //# of a MAX_PUSH_ID frame on any other stream MUST be treated as a
                 //# connection error of type H3_FRAME_UNEXPECTED.
-                Some(_) => return Err(Code::H3_FRAME_UNEXPECTED.into()),
-                None => return Ok(None),
+                Some(_) => return Poll::Ready(Err(Code::H3_FRAME_UNEXPECTED.into())),
+                None => return Poll::Ready(Ok(None)),
             }
         };
-
         if !self.stream.is_eos() {
             // Get the trailing frame
-            let trailing_frame = future::poll_fn(|cx| self.stream.poll_next(cx))
-                .await
+            let trailing_frame = futures_util::ready!(self.stream.poll_next(cx))
```

Comment:
I think this introduces a bug. There is no error returned for the case that 2 `Headers` frames are sent by the peer and the first call of `poll_next`  returns `Poll::Pending`.

Reply:
Fixed by saving the trailers when pending is encountered for the trailing_frame.
The first header frame is always used for parsing the trailers. If peer sends more header frames, we poll it until it is received and checked. This assumes is_eos() is still false after poll_next returned pending.